### DISCUSSION TITLE
Converting a Json Into a TreeView

### DISCUSSION THREAD

**Discussion by Vitco96** [AvaloniaUI/Avalonia#9995] [Q&A]
Hello everyone . 

I am currently working on a project that involves converting  an app in such a way that it works for Mac. The app was originally written in WPF and I chose to use Avalonia UI for the it. Currently i am working on a key feature that involves converting a Json into a treeview. The Json looks like this 
{
  "1_u16": 3,
  "2_astruct": [
    {
      "1_u16": 4575,
      "2_au8": [
        65,
        117,
        103,
        32,
        32,
        57,
        32,
        50,
        48,
        49,
        56,
        0
      ],
      "3_au8": [
        49,
        52,
        58,
        50,
        53,
        58,
        52,
        52,
        0
      ]
    },
    {
      "1_u16": 25205,
      "2_au8": [
        65,
        117,
        103,
        32,
        32,
        57,
        32,
        50,
        48,
        49,
        56,
        0
      ],
      "3_au8": [
        49,
        52,
        58,
        50,
        53,
        58,
        52,
        52,
        0
      ]
    },
    {
      "1_u16": 25268,
      "2_au8": [
        65,
        117,
        103,
        32,
        32,
        57,
        32,
        50,
        48,
        49,
        56,
        0
      ],
      "3_au8": [
        49,
        52,
        58,
        50,
        53,
        58,
        52,
        52,
        0
      ]
    }
  ]
}
Can anyone help ?

**Comment by timunie** [comment]
I'd use the MVVM approach where I would have a `ViewModel` and bind to the items. Check out ControlCatalog for reference: 

https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/TreeViewPage.xaml
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/ViewModels/TreeViewPageViewModel.cs

... And welcome :-)

**Reply by Vitco96** [reply]
Thank you ! 

From the server I receive a SortedDictionary<ushort, Instance> in my view model.  The Instance object has a property of type public dynamic Json. In this Json i have the hirearchy which I need to create. If the key contains the "struct" word that means it is a node  and if it does not contain the struct word it means it is a leaf.  In the old project when they found a node they assigned a new List<TreeNode>(); value to it. TreeNode is a class in the System.Windows.Forms namespace. Now it seems that a couple of objects and properties and classes that are implemented in WPF are missing from Avalonia which is giving me a big headache :)).

**Reply by timunie** [reply]
`System.Windows.Forms` is not for WPF normally. Was your App a WinForms app before? If so, you really need to get used to the MVVM way to do things.

**Reply by timunie** [reply]
Maybe see also #9927
